src/main/java/com/google/sps/data/OrganizationUpdater.java

```diff
@@ -127,15 +133,23 @@ private void setOrganizationProperty(String propertyKey, String formValue) {
     }
   }
 
-  private ArrayList<String> translateEmailsToIds(ArrayList<String> emails) {
-    ArrayList<String> userIds = new ArrayList<String>();
-    
-    // TODO(): Implement correct way of getting user by email once that is decided on
+  // Based on the regex parsed list of emails, method will add appropriate userIds or userEmails to moderatorIds or invitedModeratorEmails, respectively.
+  private void retrieveModeratorListAndInvitedModerators(ArrayList<String> emails, ArrayList<String> moderatorIds, ArrayList<String> invitedModeratorEmails) {
     for(String email : emails) {
       GivrUser newUser = GivrUser.getUserByEmail(email);
-      userIds.add(newUser.getUserId());
+      String userId = newUser.getUserId();
+      // UserId can equal "" if that user has never logged in. User email will be added to the invitedModerators list, and not the moderatorList.
+      if (userId.equals("")) {
+        if (this.entity.getProperty("invitedModerators") == null) {
+          invitedModeratorEmails = new ArrayList<String>();
+        } else {
+          invitedModeratorEmails = (ArrayList) this.entity.getProperty("invitedModerators");
+        }
```

Comment:
The way java arguments work (references by value), this is not modifying the "invitedModeratorEmails" that was passed in!

Ex:
public class Experiment {
    public static void main(String args[]) {
      String s = "In";
      foo(s);
      System.out.println(s);
    }
    public static void foo(String s) {
        s = "Out";
    }
}

Output:
"In"

Reply:
I refactored this to call two separate functions to find invited moderators vs active moderators and return the appropriate lists!